### PR TITLE
Notebook server extension to Kernel Gateway

### PR DESCRIPTION
**Pull request by jtyberg**
Notebook extension that uses remote kernels served up by Kernel Gateway.

### CODE REVIEW

**Comment by parente** [comment]
This is looking good. When you're ready for me to test from install through use, take the WIP off and let me know.

**Comment by jtyberg** [comment]
I think this is ready for a test drive.

File: nb2kg/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Jupyter Notebook Extension to Kernel Gateway (nb2kg)
+
+## Overview
+
+**nb2kg** is a [Jupyter Notebook](https://github.com/jupyter/notebook) [server extension](http://jupyter-notebook.readthedocs.io/en/latest/extending/handlers.html#writing-a-notebook-server-extension)
```
Comment (parente) [inline nb2kg/README.md]: Make sure you mention the word "demo" in here.

File: nb2kg/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+```
+jupyter serverextension disable --py nb2kg --sys-prefix
+pip uninstall -y nb2kg
+```
+
+## Caveats
+
+The **nb2kg** extension is currently demo-level code.  It is not ready for use in production environments.  To make it production-ready would require additional testing (and, ehem, test cases), as well as some changes to the [jupyter/notebook](https://github.com/jupyter/notebook) code base.  In particular,
+
+* The Notebook kernel API request handlers would need to support asynchronous operation for ALL kernel requests.  The extension currently must replace several manager and handler classes to add this support.
```
Comment (parente) [inline nb2kg/README.md]: A bullet reminding people that files appearing in the notebook dashboard are local to the notebook server but the remote KG won't have access to them.

**Comment by parente** [comment]
:shipit:

**Pull request merged**